pkgs/by-name/me/mednafen/package.nix

```diff
@@ -1,33 +1,39 @@
-{ lib
-, stdenv
-, fetchurl
-, SDL2
-, SDL2_net
-, alsa-lib
-, flac
-, freeglut
-, libGL
-, libGLU
-, libX11
-, libcdio
-, libjack2
-, libsamplerate
-, libsndfile
-, pkg-config
-, zlib
-, libiconv
+{
+  lib,
+  SDL2,
+  SDL2_net,
+  alsa-lib,
+  fetchurl,
+  flac,
+  freeglut,
+  libGL,
+  libGLU,
+  libX11,
+  libcdio,
+  libiconv,
+  libjack2,
+  libsamplerate,
+  libsndfile,
+  pkg-config,
+  stdenv,
+  zlib,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "mednafen";
   version = "1.29.0";
 
   src = fetchurl {
-    url = "https://mednafen.github.io/releases/files/${pname}-${version}.tar.xz";
+    url = "https://mednafen.github.io/releases/files/mednafen-${finalAttrs.version}.tar.xz";
     hash = "sha256-2j+88Ch3+b4PAov6XRy1npU6QEm5D+fjk4ijOG2fNi4=";
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  outputs = [ "out" "doc" ];
+
+  nativeBuildInputs = [
+    SDL2
```

Comment:
Why is `SDL2` in `nativeBuildInputs` if it is in `buildInputs`?

Reply:
When using `strictDeps = true`, we need two SDLs: the one that runs `sdl-config` from building platform and the other that installs SDL libs to target platform.

https://discourse.nixos.org/t/always-be-cross-compiling-slides-by-matthew-bauer-pdf/30287

Reply:
How exactly?

If in the sense of "it works on my machine, so it is not needed", then it is not needed.

If in the sense of "bad with it, worse without it", it is seriously needed.

Among many things, `strictDeps` enforces a cleaner separation of rules and concerns among inputs and phases - a thing that is easily blurred in native compilation but can be very tricky in cross compilation.

https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-reference

https://github.com/NixOS/nixpkgs/issues/178468

Reply:
Understandable, I wasn't aware that this was an issue with `nativeBuildInputs` and `buildInputs`.